src/core/main.c

```diff
@@ -1865,6 +1867,14 @@ static int initialize_runtime(
                 }
         }
 
+        if (arg_system && arg_no_new_privs) {
+                r = prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0);
+                if (r < 0) {
+                        *ret_error_message = "Failed to disable new privileges";
+                        return r;
```

Comment:
prctl() returns errors in errno. Needs to be hence: "return -errno" here.

That said, "return log_emergency_errno(errno, …" would be even better